src/components/Form/FormDatePicker.react.js

```diff
@@ -0,0 +1,176 @@
+    const { onChange } = this.props;
+    const newDate: Date = new Date(currentDate);
+
+    // Change month
+    if (type === "mm") {
+      newDate.setMonth(value);
+    }
+
+    // Change day
+    if (type === "dd") {
+      newDate.setDate(value);
+    }
+
+    if (type === "yyyy") {
+      newDate.setFullYear(value);
+    }
+
+    this.setState({ currentDate: newDate }, () => {
+      onChange && onChange(this.state.currentDate);
+    });
+  };
+
+  // Creates an array with numeric values from start to end
+  _range = (start: number, end: number): Array<number> =>
+    Array.from({ length: end + 1 - start }, (v, k) => k + start);
+
+  // Renders the months select
+  _renderMonths = (): React.Node => {
+    const { currentDate } = this.state;
+    const { monthLabels } = this.props;
+
+    const onChangeMonths = ({ target }) =>
+      this._handleOnChange("mm", target.value);
+
+    return (
+      <FormSelect onChange={onChangeMonths}>
+        {monthLabels &&
+          monthLabels.map((name, index) => (
+            <option
+              value={index}
+              selected={currentDate.getUTCMonth() === index}
+            >
+              {name}
+            </option>
+          ))}
+      </FormSelect>
+    );
+  };
+
+  // Renders the days select
+  _renderDays = (): React.Node => {
+    const { currentDate } = this.state;
+    const currentMonthDays = new Date(
+      currentDate.getUTCFullYear(),
+      currentDate.getUTCMonth() + 1,
+      0
+    ).getDate();
+    const daysRange = this._range(1, currentMonthDays);
+    const currentDay = currentDate.getUTCDate();
+
+    const onChangeDays = ({ target }) =>
+      this._handleOnChange("dd", target.value);
+
+    return (
+      <FormSelect onChange={onChangeDays}>
+        {daysRange.map(day => (
+          <option value={day} selected={currentDay === day}>
+            {day}
+          </option>
+        ))}
+      </FormSelect>
+    );
+  };
+
+  // renderes the years select
+  _renderYears = (): React.Node => {
+    const { minYear, maxYear } = this.props;
+    const { currentDate } = this.state;
+    const yearsRange = this._range(minYear, maxYear).reverse();
+    const currentYear = currentDate.getUTCFullYear();
+
+    const onChangeYears = ({ target }) =>
+      this._handleOnChange("yyyy", target.value);
+
+    return (
+      <FormSelect onChange={onChangeYears}>
+        {yearsRange.map(year => (
+          <option value={year} selected={currentYear === year}>
+            {year}
+          </option>
+        ))}
+      </FormSelect>
+    );
+  };
+
+  render(): React.Node {
+    const { format, className } = this.props;
+    const formatSplit = format.split("/");
+    const dateComponents = {
+      mm: this._renderMonths(),
+      dd: this._renderDays(),
+      yyyy: this._renderYears(),
+    };
+
+    return (
+      <div className={className}>
+        <FormInputGroup>
+          {formatSplit && formatSplit.map(type => dateComponents[type])}
```

Comment:
Since format is a required prop, `formatSplit` should never be null / falsy (I think) and thus we can get rid of `{formatSplit &&`